### ISSUE TITLE
No documentation for programmically setting config values

### ISSUE THREAD

**Issue by gasia89** [Azure/AppConfiguration#158]
In the Azure Portal in the 'Access Keys', section there are 2 types of keys, read keys, and read/write keys. However, nowhere in the documentation does it say anything about programmatically setting configuration values.

![AppConfiguration_1](https://user-images.githubusercontent.com/17482659/66278740-f071b900-e860-11e9-9d6b-5f8849ff21f0.png)

**Comment by zhenlan** [comment]
I think this is about the documentation of using SDK.

@gasia89 I agree this is not clear in our current documentation (https://aka.ms/AzureAppConfiguration). Before we make it better, you can find SDK information under `Reference`->`Configuration`->`.NET/Java/Python/Javascript Azure SDK`.

cc: @lisaguthrie

**Comment by gasia89** [comment]
@zhenlan I looked at the documentation you recommended, however, my question remains. All of that information is related to pulling information down from Azure. We have a key section, that is labled as a "Read-write keys", and a key section that is labled as, "Read-only keys". Let me be more specific in my question.

 What is the functional difference between the "Read-write Keys" and the "Read-only Keys"? 

As far as I can tell from the documentation I could find, and what you recommended, the connection is mostly one-way, in the sense that the client can request the current configuration as key value pairs from the server and toggle features on and off, but what if I wanted to programatically set a config value?

The name "Read-write keys" implies that I should be able to send information to the server as well. 

Poor use case, but as a simple to illistrate example: 

Say I have a webapp that is my company's home brew on-call alerting app. I want it to have an input that will allow a user to add a value to the config with their email address to notify if its their turn to be on call. I want the user to be able to enter their info, and if valid, have it pushed to the configuration manager, so that the next time the configuration refreshes, any applications watching that key, will update to include the new on call member.

As far as I can find, there is no documentation anywhere on how to do this, and the methods and props on the objects such as AzureAppConfigurationOptions, and AzureAppConfigurationExtensions, do not have anything to indicate that sending updates to the server is an option.

So if the intention is that "read-write keys" will eventually be able to write to the server, or I am grossly misunderstanding things, or the documentation is just not available, and we can do this already, then this is an appropriate name, otherwise, if there is no intention of adding that funtionality, then I feel that the name is misleading as there is no real way to set keys programatically as far as I can tell. Again, if I am incorrect here, I apologize, and please correct me.

![image](https://user-images.githubusercontent.com/17482659/66716526-728f3f80-ed83-11e9-96cc-2ce23a8e290e.png)

**Comment by zhenlan** [comment]
Hi @gasia89, I was not very sure about your question. Thanks for the clarification.

The short answer is, yes, you can already write config to the server programmatically using SDKs (besides using REST APIs).

Here goes the long answer :)
#### Access keys
- *Read-write keys* can be used for both reading and writing configuration from/to an App Configuration store.
- *Read-only keys* can only be used for reading configuration from an App Configuration store. The operation will fail if it's used for writing configuration to an App Configuration store.

#### Configuration Providers
- *Configuration providers* is a programming model in .NET Core/ASP.NET Core for *reading* configuration data into key-value pairs from a variety of configuration sources.
- `Microsoft.Extensions.Configuration.AzureAppConfiguration` is a configuration provider for *reading* configuration from the Azure App Configuration.

#### Azure App Configuration SDKs
- SDKs are for the generic operations (read, write, delete, etc.) with the Azure App Configuration. Their APIs normally can be mapped to REST APIs of the service. SDKs are available in different programming languages but, unlike configuration providers, they are *not* tailored for any specific programming model.
- [Here](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/appconfiguration/Azure.Data.AppConfiguration#examples) are some .NET code samples of creating/retrieving/updating/deleting a configuration setting in an App Configuration store. Please note you will need the read-write connection string for all but the retrieving operation.

Hope this helps. Again, we should make this more obvious in our documentation.

**Comment by gasia89** [comment]
Perfect! That was exactly the documentation I needed. I was not thinking about this from the context of it being a configuration provider. Once you said that it made more sense why this works this way. 

I do agree that the documentation could use some clarity on this topic. I did not see any mention of the sdk in other documentation, and I did delve into the readme a bit. I did not see the examples page you linked on my intial look through of said readme, clearly I missed it. 

It might even be worth it to just reference that link from the existing documentation. That would at least provide a quick reference for others facing similar confusion. Thanks again for your help!
